scrowl-io/scrowl
# sc-977 - Feature - Application menu create/save/save as

### Short summary

This PR includes features from three different stories:

[[sc-977]](https://app.shortcut.com/osgca/story/977/create-a-new-course-project) Create a new course project
[[sc-981]](https://app.shortcut.com/osgca/story/981/save-as-a-course-project) "Save As" a course project
[[sc-1046]](https://app.shortcut.com/osgca/story/1046/save-a-project) Save a course project

---

### Test steps

1. Run `yarn start:tool:course`
2. Go to the application menu `File > New Project...` to create a new project. Check the dev tools console log to find the new project temp directory.
3. Go to the application menu `File > Save` to save the project in a `.scrowl` file.
4. Go to the application menu `File > Save As...` to save the file in a different location than the previous save.
5. Restart the application and repeat the test using keyboard shortcuts `command + N` to create the project, `command + S` to save the project, `command + shift + S` to save it in a different location.
